x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/job/FeatureIndexBuilderJobConfig.java

```diff
@@ -28,25 +30,55 @@
 
     private static final String NAME = "xpack/feature_index_builder/jobconfig";
     private static final ParseField ID = new ParseField("id");
+    private static final ParseField INDEX_PATTERN = new ParseField("index_pattern");
+    private static final ParseField DESTINATION_INDEX = new ParseField("destination_index");
+    private static final ParseField SOURCES = new ParseField("sources");
+    private static final ParseField AGGREGATIONS = new ParseField("aggregations");
 
     private final String id;
+    private final String indexPattern;
+    private final String destinationIndex;
+    private final SourceConfig sourceConfig;
+    private final AggregationConfig aggregationConfig;
 
     private static final ConstructingObjectParser<FeatureIndexBuilderJobConfig, String> PARSER = new ConstructingObjectParser<>(NAME, false,
```

Comment:
Experience has shown that for stored configs it's best to have both a lenient parser that ignores unknown fields and a strict parser that throws an error on encountering an unknown field.  See for example `AnalysisConfig.LENIENT_PARSER` and `AnalysisConfig.STRICT_PARSER`.  Then use the strict parser when parsing user-supplied requests and the lenient parser when parsing configs retrieved from storage.  That way things don't completely break in a mixed version cluster when the newer version has added an extra field to the config.

Reply:
made a note int the meta issue (https://github.com/elastic/elasticsearch/issues/33600), as we will highly likely move out of the cluster state (https://github.com/elastic/elasticsearch/issues/33952) the config needs a full revisit